File: 06.UITableViewCell/Sample Code/CellStylesObjc/CellStylesObjc/ViewController.m
#import "ViewController.h"

@interface ViewController () <UITableViewDataSource, UITableViewDelegate>

@property (weak, nonatomic) IBOutlet UITableView *tableView;

@end

@implementation ViewController

#pragma mark - UITableViewDataSource Methods

- (NSInteger)tableView:(UITableView *)tableView
 numberOfRowsInSection:(NSInteger)section
{
    return 5;
}

- (UITableViewCell *)tableView:(UITableView *)tableView
         cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = nil;
    
    switch ([indexPath row])
    {
        case 1:
            cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleValue1 reuseIdentifier:@"Value1"];
            [cell setAccessoryType:UITableViewCellAccessoryDisclosureIndicator];
            break;
        case 2:
            cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleValue2 reuseIdentifier:@"Value2"];
            [cell setAccessoryType:UITableViewCellAccessoryDetailDisclosureButton];
            break;
        case 3:
            cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:@"Subtitle"];
            [cell setAccessoryType:UITableViewCellAccessoryCheckmark];
            break;
        case 4:
            cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleSubtitle reuseIdentifier:@"Subtitle2"];
            [[cell textLabel] setText:@"Hello"];
            [cell setAccessoryType:UITableViewCellAccessoryDetailButton];
            return cell;
            break;
        default:
            cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"Default"];
            [cell setAccessoryType:UITableViewCellAccessoryNone];
            break;
    }
    
    [[cell textLabel] setText:@"Hello"];
    [[cell detailTextLabel] setText:@"World"];
    [cell setAccessoryView:[[UISwitch alloc] init]];
    [[cell imageView] setImage:[UIImage imageNamed:@"image"]];
    
    return cell;
}

@end

File: 06.UITableViewCell/Sample Code/CellStylesSwift/CellStylesSwift/ViewController.swift
import UIKit

class ViewController: UIViewController, UITableViewDataSource, UITableViewDelegate
{
    @IBOutlet weak var tableView: UITableView!
    
    // MARK: - UITableViewDataSource Methods
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int
    {
        return 5
    }

    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell
    {
        var cell: UITableViewCell
        
        switch(indexPath.row)
        {
        case 1:
            cell = UITableViewCell(style: .Value1, reuseIdentifier: "Value1")
            cell.accessoryType = .DisclosureIndicator
        case 2:
            cell = UITableViewCell(style: .Value2, reuseIdentifier: "Value2")
            cell.accessoryType = .DetailDisclosureButton
        case 3:
            cell = UITableViewCell(style: .Subtitle, reuseIdentifier: "Subtitle")
            cell.accessoryType = .Checkmark
        case 4:
            cell = UITableViewCell(style: .Subtitle, reuseIdentifier: "Subtitle2")
            cell.textLabel?.text = "Hello"
            cell.accessoryType = .DetailButton
            return cell
        default:
            cell = UITableViewCell(style: .Default, reuseIdentifier: "Default")
            cell.accessoryType = .None
        }
        
        cell.textLabel?.text = "Hello"
        cell.detailTextLabel?.text = "World"
        cell.accessoryView = UISwitch()
        cell.imageView?.image = UIImage(named: "image")

        return cell
    }
}

